packages/vats/src/core/startWalletFactory.js

```diff
@@ -238,18 +217,45 @@ export const startWalletFactory = async (
       economicCommitteeCreatorFacet,
     },
   );
+  provisionPoolStartResult.resolve(ppFacets);
   instanceProduce.provisionPool.resolve(ppFacets.instance);
 
-  provisionPoolStartResult.resolve(ppFacets);
+  const terms = await deeplyFulfilled(
+    harden({
+      agoricNames,
+      board,
+      assetPublisher: Far('AssetPublisher', {
+        getAssetSubscription: () => E(poolBank).getAssetSubscription(),
+      }),
+    }),
+  );
+  /** @type {WalletFactoryStartResult} */
+  const wfFacets = await E(zoe).startInstance(
+    walletFactory,
+    { Fee: feeIssuer },
+    terms,
+    {
+      storageNode: walletStorageNode,
+      walletBridgeManager,
+      walletReviver: ppFacets.walletReviver,
+    },
+    'walletFactory',
+  );
+  walletFactoryStartResult.resolve(wfFacets);
+  instanceProduce.walletFactory.resolve(wfFacets.instance);
 
-  const handler = await E(ppFacets.creatorFacet).makeHandler({
-    bankManager,
-    namesByAddressAdmin,
-    walletFactory: wfFacets.creatorFacet,
-  });
+  await Promise.all([
+    E(ppFacets.creatorFacet).addRevivableAddresses(oldAddresses),
+    E(ppFacets.creatorFacet).setReferences({
+      bankManager,
+      namesByAddressAdmin,
+      walletFactory: wfFacets.creatorFacet,
+    }),
```

Comment:
Are these uses of `ppFacets` (along with `walletReviver: ppFacets.walletReviver` above) correct? The tests had to use `E(facets.creatorFacet).getLimitedCreatorFacet()`, but after starting in a different way (`const facets = await E(zoe).startInstance(…)` rather than `const ppFacets = await startGovernedInstance(…)` as is the case here).